backend/src/schema/index.js

```diff
@@ -12,11 +12,25 @@ export default applyScalars(
       resolvers,
       config: {
         query: {
-          exclude: ['InvitationCode', 'EmailAddress', 'Notfication', 'Statistics', 'LoggedInUser'],
+          exclude: [
+            'Badge',
+            'InvitationCode',
+            'EmailAddress',
+            'Notfication',
+            'Statistics',
+            'LoggedInUser',
+          ],
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-07-11T14:13:23Z" title="Thursday, July 11th 2019, 4:13:23 pm +02:00">Jul 11, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -12,11 +12,25 @@ export default applyScalars(
       resolvers,
       config: {
         query: {
-          exclude: ['InvitationCode', 'EmailAddress', 'Notfication', 'Statistics', 'LoggedInUser'],
+          exclude: [
+            'Badge',
+            'InvitationCode',
+            'EmailAddress',
+            'Notfication',
+            'Statistics',
+            'LoggedInUser',
+          ],
```


we're excluding these because we are not using `neo4j-graphql-js` for these, correct?

Reply:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-07-11T14:14:23Z" title="Thursday, July 11th 2019, 4:14:23 pm +02:00">Jul 11, 2019</time>_

---


right, so I see we are using `neo4j-graphql-js` for the query, so why is it excluded here?

Reply:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-07-11T14:24:09Z" title="Thursday, July 11th 2019, 4:24:09 pm +02:00">Jul 11, 2019</time>_

---


as I understood it, this is part of excluding it from the auto-generated queries when passed into `makeAugmentedSchema`, which according to the `neo4j-graphql-js` docs is `Commonly used to exclude payload types.`
https://grandstack.io/docs/neo4j-graphql-js-api.html
What is the purpose of excluding `Badge` here?
If we use `neo4j-graphql-js` in `backend/src/schema/resolvers/badges.js` aren't we using the same query, or are there others that are auto-generated that you want to blacklist?

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-07-11T19:47:00Z" title="Thursday, July 11th 2019, 9:47:00 pm +02:00">Jul 11, 2019</time>_

---


If you exclude types here, that means `neo4j-graphql-js` is not *augmenting* the schema to add queries/mutations. Nothing stops us from adding the desired queries to the schema ourselves. If we follow the naming schema, `neo4jgraphqljs` works fine in the resolver.

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-07-11T19:47:33Z" title="Thursday, July 11th 2019, 9:47:33 pm +02:00">Jul 11, 2019</time>_

---


This method uses the `resolveInfo` as input, which has all the information about the requested query or mutation and all the params.